feat: add mlflow as default

linusseelinger opened the pull request:
#### Description of changes
We provide an MLflow server in P4D now, passing the `TESSERACT_MLFLOW_TRACKING_URI` env var to **every** Tesseract. With https://github.com/pasteurlabs/AutoPhysics/pull/127, AP follows suit.

This leads to an issue atm if Tesseracts don't explicitly include the mlflow package as a dependency; they receive the env var, switch to mlflow backend, and fail to find the package.

This PR includes mlflow by default in Tesseracts (including cleanups around that). We use the `mlflow-skinny` package to keep dependencies and complexity minimal. This is possible since we only need a simple mlflow client.

(An alternative could be graceful failure in the scenario above. In that case, it may not be immediately clear to a user why mpa logs of some Tesseracts don't end up in P4D's/AP's MLflow instance. Since have now decided on MLflow as the technology of choice for metrics/params in our ecosystem, we might as well make Tesseracts "batteries included" in that regard.)

Additionally, we provide an actual mlflow server instance during testing (authored by @dionhaefner ).

#### Testing done
Local testing, CI

apaleyes reviewed (approved):
looks pretty clean, thanks!

linusseelinger commented on tests/runtime_tests/test_mpa.py:
```diff
@@ -141,84 +180,45 @@ def test_log_artifact_missing_file():
+
+def test_mlflow_run_extra_args(mocker, dummy_mlflow_server):
     """Test passing a dict with basic tags."""
     pytest.importorskip("mlflow")
```
I think we still support/need `mlflow_run_extra_args`. Cleaned up a bit though.

dionhaefner commented:
> always map to port 5000 (@dionhaefner was it intentional to map to an ephemeral port instead?)

Yes that was intentional since 5000 may be taken. Hence why I added a network and service name so you can address the container by name instead.

linusseelinger commented:
> > always map to port 5000 (@dionhaefner was it intentional to map to an ephemeral port instead?)
> 
> Yes that was intentional since 5000 may be taken. Hence why I added a network and service name so you can address the container by name instead.

Makes sense! I reverted my change and added usage docs for the ephemeral port / docker network.

dionhaefner commented:
Please fix the test by not using the Docker fixture in the base test suite. Should be no reason to spawn a full MLFlow server to test this interaction.
